Sources/SwiftProtobuf/_MessageStorage+BinaryDecoding.swift

```diff
@@ -398,7 +404,116 @@ extension _MessageStorage {
+            field,
+            self,
+            isRepeated ? .append : .mutate
+        ) { outRawValue in
+            // In the singular case, this doesn't matter. In the repeated case, we need to return
+            // true *exactly once* and then return false the next time this is called. This is
+            // because the same trampoline function is used to handle the packed case, where it
+            // calls the closure over and over until it returns that there is no data left.
+            guard !alreadyReadValue else { return false }
+            outRawValue = Int32(bitPattern: UInt32(truncatingIfNeeded: try reader.nextVarint()))
+            alreadyReadValue = true
+            return true
+        } /*onInvalidValue*/ _: { rawValue in
+            // Serialize the invalid values into a binary blob that will be passed as a single
+            // varint field into unknown fields.
+            //
+            // Note that because we've already read the value, we have to put it into unknown fields
+            // ourselves. The `decodeUnknownField` flow assumes that we've only read the tag and
+            // are positioned at the beginning of the value.
+            let fieldTag = FieldTag(fieldNumber: fieldNumber, wireFormat: .varint)
+            let fieldSize = fieldTag.encodedSize + Varint.encodedSize(of: Int64(rawValue))
+            var field = Data(count: fieldSize)
+            field.withUnsafeMutableBytes { body in
+                var encoder = BinaryEncoder(forWritingInto: body)
+                encoder.startField(tag: fieldTag)
+                encoder.putVarInt(value: Int64(rawValue))
+                unknownFields.append(protobufBytes: UnsafeRawBufferPointer(body))
+            }
+        }
+    }
+
+    /// Appends either a single enum value or multiple packed values to the array value of the
+    /// given field, depending on the wire format of the corresponding tag.
+    ///
+    /// - Parameters:
+    ///   - reader: The reader from which the values to append should be read.
+    ///   - field: The field being decoded.
+    ///   - tag: The tag that was read from the wire.
+    private func appendPackedEnumValues(
+        from reader: inout WireFormatReader,
+        to field: FieldLayout,
+        fieldNumber: Int
+    ) throws {
+        assert(field.rawFieldType == .enum, "Internal error: should only be called for enum fields")
+
+        let elementsBuffer = try reader.nextLengthDelimitedSlice()
+        guard elementsBuffer.baseAddress != nil, elementsBuffer.count > 0 else {
+            return
+        }
+
+        // If we see any invalid values during decode them, save them here so we can write them
+        // into unknown fields at the end.
+        var invalidValues: [Int32] = []
 
+        // Recursion budget is irrelevant here because we should never recurse into groups or
+        // messages here; they're not supported as packed fields.
```

Comment:
Teh "not supported in packed fields" is sorta not related, there's no submessage in a list of enums. 😄